drivers/ethernet/eth_native_posix.c

```diff
@@ -231,55 +231,134 @@ static inline struct net_if *get_iface(struct eth_context *ctx,
 #endif
 }
 
-static int read_data(struct eth_context *ctx, int fd)
+#if defined(CONFIG_NET_VLAN)
+static struct net_pkt *prepare_vlan_pkt(struct eth_context *ctx,
+					int count, u16_t *vlan_tag, int *status)
 {
-	u16_t vlan_tag = NET_VLAN_TAG_UNSPEC;
-	struct net_if *iface;
+	struct net_eth_vlan_hdr *hdr = (struct net_eth_vlan_hdr *)(ctx->recv);
```

Comment:
> (struct net_eth_vlan_hdr *)(ctx->recv);

Parens around `(ctx->recv)` are superfluous, I don't think we use those in Zephyr codestyle, so those are inconsistent.